test-requirements.txt

```diff
@@ -0,0 +1,12 @@
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+flake8==4.0.1
+black
+flake8-copyright<0.3
+juju==3.0.1
+pytest
+pytest-mock
+pyyaml
+tenacity<8.1
+requests
```

Comment:
I believe some of these dependencies are not used. Mostly, tenacity, requests and pytest-mock. Can you double check which ones are actually required?